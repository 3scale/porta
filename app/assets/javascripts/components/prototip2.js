

var Bubble = Class.create();
Bubble.prototype = {
  initialize: function(el) {
    this.offsetX = -8;
    this.offsetY = 38;

    this.target = el;
    // this.text = text;
    this.makeBubble();
  },
 
  makeBubble: function() {
    var bubble = $("bubble_" + this.target.title);
    // var bubble = new Element('div', { 'class': 'bubble'});
    // var content = new Element('div', { 'class': 'content'});
    // content.innerHTML = this.text;

    // bubble.appendChild(content);
    this.bubble = bubble;
    // new Insertion.Bottom(body, bubble);
    // bubble.hide();
    
    
    
    // set position

    // listen!
    
    this.target.observe('mouseover', (
        function() { 

          var position = this.target.cumulativeOffset();

          this.bubble.style.top = position[1] + this.offsetX + "px";
          this.bubble.style.left = position[0] + this.target.getWidth() + this.offsetY + "px";
          
          this.bubble.show(); 
        }).bind(this));
        
    this.target.observe('mouseout', (
        function() { 
          this.bubble.hide(); 
        }).bind(this));
        
  }
  
  
}

var BubbleHelper = {
  set: function(){
    $$('.define').each(function(l){
      new Bubble(l);      
    });
}}
  //     function(event) { 
  // 
  //       if ((event.element().className == 'submit' )) {
  // 
  //         var form = event.element().up('form');
  //         var formData = form.serialize(true);
  //         var url = form.action;
  //         
  //         this.ajaxPost(formData, url);
  //         event.stop();
  //       }
  //     })
  //   )        
  //   
  // }
