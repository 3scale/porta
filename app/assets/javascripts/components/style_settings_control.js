// Functions dealing with provider side control of site colours. Probably to be deprecated with the
// development of CMS system

jQuery.noConflict();

var settings = function() {
    // var B = ["background_color", "text_color", "link_color", "sidebar_fill_color", "sidebar_border_color"];
    var A = {
        updateColors: function() {
            var E = jQuery("#settings_bg_colour").val();
            var G = jQuery("#settings_text_colour").val();
            var F = jQuery("#settings_link_colour").val();
            var C = jQuery("#settings_menu_bg_colour").val();
						var D = jQuery("#settings_menu_link_colour").val();
						var H = jQuery("#settings_content_bg_colour").val();
						var J = jQuery("#settings_plans_tab_bg_colour").val();
						var K = jQuery("#settings_content_border_colour").val();
						
            jQuery("#previewContent").css({"background-color": E, color: G});

						jQuery('#pBox span').css("color", G);
						jQuery('#pBox  span a').css("color", F);
						jQuery('#pLinks span.active').css("background-color", H);
						jQuery('#pLinks span').css("color", D);
						jQuery('#pBox, #pFooter').css("background-color", H);
						jQuery('#pBox, #pFooter').css("border-color", K);
						jQuery('#previewPlans table th.head').css("background-color", J);
						
            jQuery("#pMenu").css({
                "background-color": C
            });
        },
        synchColors: function(D) {
            var C = settings.colorPicker.linked;
            if (C && C.value && C.value != settings.colorPicker.color) {
                C.value = settings.colorPicker.color;
                jQuery(C).css({
                    "background-color": C.value,
                    color: settings.colorPicker.hsl[2] > 0.5 ? "#000": "#fff"
                });
                settings.updateColors()
            }
        },
        initDesignSettings: function() {

          jQuery("#linkDesign a").click(function() {
              jQuery("#designSettings").show();
							jQuery('#detailSettings').hide();
              jQuery("#linkDetails a").removeClass("active");
              jQuery("#linkDesign a").addClass("active");
              jQuery("#linkDetails").removeClass("active");
              jQuery("#linkDesign").addClass("active");
							
              return false
          });

          jQuery("#linkDetails a").click(function() {
              jQuery("#designSettings").hide();
							jQuery('#detailSettings').show();

              // jQuery("#backgrounds").hide();
              jQuery("#linkDetails a").addClass("active");
              jQuery("#linkDesign a").removeClass("active");
              jQuery("#linkDetails").addClass("active");
              jQuery("#linkDesign").removeClass("active");

              // if (jQuery("#colors").css("display") == "none") {
              //     jQuery("#title_picker").removeClass("active");
              //     jQuery("#title_background").removeClass("inactive")
              // } else {
              //     jQuery("#title_picker").addClass("active");
              //     jQuery("#title_background").addClass("inactive")
              // }
              // if (jQuery("#user_profile_default").val()) {
              //     jQuery("#user_profile_default").val(false)
              // }
              return false
          });

            jQuery("#title_picker a").click(function() {
                jQuery("#colors").toggle();
                jQuery("#backgrounds").hide();
                jQuery("#title_background").removeClass("active");
                if (jQuery("#colors").css("display") == "none") {
                    jQuery("#title_picker").removeClass("active");
                    jQuery("#title_background").removeClass("inactive")
                } else {
                    jQuery("#title_picker").addClass("active");
                    jQuery("#title_background").addClass("inactive")
                }
                if (jQuery("#user_profile_default").val()) {
                    jQuery("#user_profile_default").val(false)
                }
                return false
            });
            jQuery("#title_background a").click(function() {
                jQuery("#backgrounds").toggle();
                jQuery("#colors").hide();
                jQuery("#title_picker").removeClass("active");
                if (jQuery("#backgrounds").css("display") == "none") {
                    jQuery("#title_background").removeClass("active");
                    jQuery("#title_picker").removeClass("inactive")
                } else {
                    jQuery("#title_background").addClass("active");
                    jQuery("#title_picker").addClass("inactive")
                }
                return false
            });
            jQuery("#themes a").click(function() {
                var E = jQuery(this).attr("id");
                settings.setTheme(E.substring(5));
                return false
            });
            jQuery("#backgrounds a").click(function() {
                jQuery("#backgrounds a").removeClass("active");
                jQuery(this).addClass("active");
                var E = jQuery(this).attr("id");
                if (E == "current_background") {
                    settings.setBackgroundImage(_tmp.currentImage)
                } else {
                    if (E == "no_background") {
                        settings.setBackgroundImage("none")
                    }
                }
                return false
            });
            jQuery("#user_profile_background_tile").click(function() {
                var E = jQuery("#user_profile_background_tile:checked").val();
                jQuery("body").css({
                    "background-repeat": E ? "repeat": "no-repeat",
                    "background-attachment": E ? "scroll": "fixed"
                });
                jQuery("#user_profile_use_background_image").val("true")
            });
            settings.colorPicker = jQuery.farbtastic("#picker", settings.synchColors);
            jQuery("#picker").click(function() {
                jQuery(".colorwell-selected").focus()
            });
            var D;
            var C = (function() {
                var E = [8, 9, 13, 16, 17, 18, 37, 38, 39, 40, 45, 46, 86, 88, 90];
                for (var F = 48; F <= 57; F++) {
                    E.push(F)
                }
                for (var F = 65; F <= 70; F++) {
                    E.push(F)
                }
                for (var F = 96; F <= 105; F++) {
                    E.push(F)
                }
                return E
            })();
            jQuery(".colorwell").each(settings.synchColors).focus(function() {
                if (D) {
                    jQuery(D).removeClass("colorwell-selected")
                }
                settings.colorPicker.linked = this;
                settings.colorPicker.setColor(this.value);
                jQuery("#picker").show();
                jQuery(D = this).addClass("colorwell-selected")
            }).blur(function() {
                var E = jQuery(E);
                E.val(E.val().toUpperCase())
            }).keydown(function(E) {
                if (jQuery.inArray(E.keyCode, C) != -1) {
                    if (jQuery.inArray(E.keyCode, [86, 88, 90]) != -1) {
                        return (E.ctrlKey || E.metaKey)
                    } else {
                        if ((jQuery.inArray(E.keyCode, [8, 46]) != -1) || ((E.keyCode == 88) && E.shiftKey)) {
                            return (jQuery(this).val() != "#")
                        } else {
                            if (E.shiftKey) {
                                return (jQuery.inArray(E.keyCode, [65, 66, 67, 68, 69, 70]) != -1)
                            }
                        }
                    }
                    return true
                }
                return false
            }).keyup(function(E) {
                if (jQuery(this).val().indexOf("#") != 0) {
                    jQuery(this).val("#" + jQuery(this).val())
                }
                settings.colorPicker.setColor(this.value);
                var F = jQuery(this).val();
                if (!document.all || (F.length == 7 || F.length == 4)) {
                    jQuery(this).css({
                        "background-color": jQuery(this).val(),
                        color: settings.colorPicker.hsl[2] > 0.5 ? "#000000": "#ffffff"
                    });
                    settings.updateColors()
                }
            });
            jQuery("#design_customization form a").click(settings.updateColors)
        }
    };
    return A
} ();

jQuery(document).ready(function() {
		if($('settingsBox')) 
    	settings.initDesignSettings();
});
