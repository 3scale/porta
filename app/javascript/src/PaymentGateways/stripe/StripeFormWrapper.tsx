/* eslint-disable react/jsx-props-no-spreading */

import { StripeElementsForm } from 'PaymentGateways/stripe/StripeElementsForm'
import { StripeElementsProvider } from 'PaymentGateways/stripe/StripeElementsProvider'
import { createReactWrapper } from 'utilities/createReactWrapper'

type Props = React.ComponentProps<typeof StripeElementsForm> & React.ComponentProps<typeof StripeElementsProvider>

const StripeFormWrapper = ({ publishableKey, ...stripeElementsFormProps }: Props, containerId: string): void => {
  createReactWrapper(
    <StripeElementsProvider publishableKey={publishableKey}>
      <StripeElementsForm {...stripeElementsFormProps} />
    </StripeElementsProvider>,
    containerId
  )
}

export { StripeFormWrapper }
