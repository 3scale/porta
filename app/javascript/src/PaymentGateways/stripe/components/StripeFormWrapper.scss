$white: white;
$black: black;
$stripe-form-card-error-color: rgb(105, 115, 134);
$stripe-form-input-placeholder-color: #3f454c;
$stripe-form-font-family: "Helvetica Neue", Helvetica, sans-serif;

.StripeElementsForm {

  .editCardButton {
    display: inline-block;
    font-size: 14px;
    font-style: italic;
    font-weight: normal;
    text-align: right;
    text-decoration: none;
    width: 100%;

    &:hover {
      text-decoration: none;
    }

    .fa {
      margin-right: 5px;
    }
  }

  .cardErrors {
    margin: 1rem;
  }

  input {
    background-color: transparent;
    border: 0;
    color: $black;
    font-family: $stripe-form-font-family;
    font-size: 19px;
    font-weight: 300;
    width: 100%;
  }

  input:focus {
    outline: none;
  }

  input::placeholder {
    color: $stripe-form-input-placeholder-color;
    font-style: normal;
  }

  .form-element + .form-element {
    margin-top: 6px;
  }

  .hidden {
    display: none;
  }

  .card-error {
    color: $stripe-form-card-error-color;
    font-size: 13px;
    line-height: 17px;
    margin-top: 12px;
    text-align: left;
  }

}
