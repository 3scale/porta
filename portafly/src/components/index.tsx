export * from './pages'
export * from './shared'
export * from './AppLayout'
export * from './AppNavExpandable'
export * from './AppNavGroup'
export * from './AppNavItem'
export * from './AppHeaderTools'
export * from './LazyRoute'
export * from './NotFound'
export * from './SwitchWith404'
export * from './util'
export * from './Root'
