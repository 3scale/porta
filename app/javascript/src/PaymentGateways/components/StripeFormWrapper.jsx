// @flow

import React, { useState } from 'react'
import {loadStripe} from '@stripe/stripe-js'
import {Elements, CardElement, useStripe, useElements} from '@stripe/react-stripe-js'
import {createReactWrapper} from 'utilities/createReactWrapper'
import {CSRFToken} from 'utilities/utils'
import 'PaymentGateways/styles/stripe.scss'

type Props = {
  stripePublishableKey: string,
  setupIntentSecret: string,
  billingAddressDetails: {
    line1: string,
    line2: string,
    city: string,
    state: string,
    postal_code: string,
    country: string
  },
  successUrl: string,
  isCreditCardStored: boolean
}

const EditCreditCardDetails = ({isCreditCardStored}: {isCreditCardStored: boolean}) => {
  const [state, setState] = useState<{isStripeFormVisible: boolean}>({ isStripeFormVisible: !isCreditCardStored })

  const setStripeFormVisible = (isStripeFormVisible: boolean) => {
    setState(prevState => ({ ...prevState, isStripeFormVisible }))
  }

  const toggleStripeForm = () => {
    const newStateStripeFormVisible = !state.isStripeFormVisible
    const stripeForm = document.getElementById('stripe-form')
    newStateStripeFormVisible ? stripeForm.classList.remove('hidden') : stripeForm.classList.add('hidden')
    setStripeFormVisible(newStateStripeFormVisible)
  }

  return (
    <a className='card-on-file' onClick={toggleStripeForm}>
      <i className='fa fa-pencil'></i>
      <span>{state.isStripeFormVisible ? 'cancel' : 'Edit Credit Card Details'}</span>
    </a>
  )
}

const CreditCardErrors = ({cardErrorMessage}: {cardErrorMessage: null | string}) => {
  if (cardErrorMessage) {
    return (
      <span id='card-errors' role='alert'>
        {cardErrorMessage}
      </span>
    )
  } else {
    return (<div></div>)
  }
}

const StripeElementsForm = ({ stripePublishableKey, setupIntentSecret, billingAddressDetails, successUrl, isCreditCardStored }: Props) => {
  const stripePromise = loadStripe(stripePublishableKey)

  const CheckoutForm = () => {
    const stripe = useStripe()
    const elements = useElements()

    const [state, setState] = useState<{cardErrorMessage: null | string}>({ cardErrorMessage: null })

    const setCardErrorMessage = (cardErrorMessage: null | string) => {
      setState(prevState => ({ ...prevState, cardErrorMessage }))
    }

    const handleSubmit = async event => {
      event.preventDefault()

      setCardErrorMessage(null)

      const {error, setupIntent} = await stripe.confirmCardSetup(setupIntentSecret, {
        payment_method: {
          card: elements.getElement(CardElement),
          billing_details: {
            address: billingAddressDetails
          }
        }
      })

      if (setupIntent && setupIntent.status === 'succeeded') {
        document.getElementById('stripe_payment_method_id').value = setupIntent.payment_method
        document.getElementById('stripe-form').submit()
      } else {
        setCardErrorMessage(error.message)
      }
    }

    const CARD_OPTIONS = {
      iconStyle: 'solid',
      style: {
        base: {
          iconColor: '#8898AA',
          color: 'black',
          fontWeight: 300,
          fontFamily: '"Helvetica Neue", Helvetica, sans-serif',
          fontSize: '19px',
          '::placeholder': {
            color: '#3f454c'
          }
        },
        invalid: {
          iconColor: '#e85746',
          color: '#e85746'
        }
      },
      classes: {
        focus: 'is-focused',
        empty: 'is-empty'
      }
    }

    return (
      <div className='well StripeElementsForm'>
        <EditCreditCardDetails isCreditCardStored={isCreditCardStored} />
        <form onSubmit={handleSubmit} action={successUrl} id='stripe-form' acceptCharset='UTF-8' method='post' className={isCreditCardStored ? 'hidden' : 'card-missing'}>
          <fieldset>
            <legend>Credit card details</legend>
            <CardElement options={CARD_OPTIONS} className='col-md-10'/>
            <CreditCardErrors cardErrorMessage={state.cardErrorMessage} />
            <span id='card-errors' role='alert'></span>
          </fieldset>
          <fieldset>
            <div className='form-group'>
              <div className='col-md-10 operations'>
                <button type='submit' disabled={!stripe} className='btn btn-primary pull-right' id='stripe-submit'>
                  Save details
                </button>
              </div>
            </div>
          </fieldset>
          <input id='stripe_payment_method_id' name='stripe[payment_method_id]' type='hidden' value=''/>
          <CSRFToken/>
        </form>
      </div>
    )
  }

  return (
    <Elements stripe={stripePromise}>
      <CheckoutForm />
    </Elements>
  )
}

const StripeFormWrapper = (props: Props, containerId: string) => createReactWrapper(<StripeElementsForm {...props}/>, containerId)

export {
  StripeFormWrapper
}
