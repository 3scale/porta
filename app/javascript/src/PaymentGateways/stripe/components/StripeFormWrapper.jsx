// @flow

import * as React from 'react'
import { loadStripe } from '@stripe/stripe-js'
import { Elements } from '@stripe/react-stripe-js'
import { createReactWrapper } from 'utilities/createReactWrapper'
import { StripeCardForm } from 'PaymentGateways'
import 'PaymentGateways/stripe/styles/stripe.scss'

type Props = {
  stripePublishableKey: string,
  setupIntentSecret: string,
  billingAddressDetails: {
    line1: string,
    line2: string,
    city: string,
    state: string,
    postal_code: string,
    country: string
  },
  successUrl: string,
  isCreditCardStored: boolean
}

const StripeElementsForm = ({ stripePublishableKey, setupIntentSecret, billingAddressDetails, successUrl, isCreditCardStored }: Props): React.Node => {
  const stripePromise = loadStripe(stripePublishableKey)

  return (
    <Elements stripe={stripePromise}>
      <StripeCardForm
        setupIntentSecret={setupIntentSecret}
        billingAddressDetails={billingAddressDetails}
        successUrl={successUrl}
        isCreditCardStored={isCreditCardStored}
      />
    </Elements>
  )
}

const StripeFormWrapper = (props: Props, containerId: string): void => (
  createReactWrapper(<StripeElementsForm {...props} />, containerId)
)

export { StripeElementsForm, StripeFormWrapper }
