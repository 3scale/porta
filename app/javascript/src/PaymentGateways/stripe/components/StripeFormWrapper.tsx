import { loadStripe } from '@stripe/stripe-js'
import { Elements } from '@stripe/react-stripe-js'
import { createReactWrapper } from 'utilities/createReactWrapper'
import { StripeCardForm } from 'PaymentGateways/stripe/components/StripeCardForm'
import 'PaymentGateways/stripe/styles/stripe.scss'

type Props = {
  stripePublishableKey: string,
  setupIntentSecret: string,
  billingAddressDetails: {
    line1: string,
    line2: string,
    city: string,
    state: string,
    // eslint-disable-next-line camelcase
    postal_code: string,
    country: string
  },
  successUrl: string,
  isCreditCardStored: boolean
}

const StripeElementsForm: React.FunctionComponent<Props> = ({
  stripePublishableKey,
  setupIntentSecret,
  billingAddressDetails,
  successUrl,
  isCreditCardStored
}) => {
  const stripePromise = loadStripe(stripePublishableKey)

  return (
    <Elements stripe={stripePromise}>
      <StripeCardForm
        billingAddressDetails={billingAddressDetails}
        isCreditCardStored={isCreditCardStored}
        setupIntentSecret={setupIntentSecret}
        successUrl={successUrl}
      />
    </Elements>
  )
}

// eslint-disable-next-line react/jsx-props-no-spreading
const StripeFormWrapper = (props: Props, containerId: string): void => createReactWrapper(<StripeElementsForm {...props} />, containerId)

export { StripeElementsForm, StripeFormWrapper, Props }
