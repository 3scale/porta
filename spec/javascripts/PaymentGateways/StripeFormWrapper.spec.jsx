import React from 'react'
import { StripeElementsForm } from 'PaymentGateways/components/StripeFormWrapper'
import { mount } from 'enzyme'

const props = {
  stripePublishableKey: 'abcd',
  setupIntentSecret: 'efgh',
  billingAddressDetails: {
    line1: '1002 Avenue de los Mexicanos',
    line2: '',
    city: 'South Park',
    state: 'CO',
    postal_code: '80440',
    country: 'US'
  },
  successUrl: '/Broflovski/Residence',
  isCreditCardStored: false
}

it('should render properly', () => {
  const wrapper = mount(<StripeElementsForm { ...props } />)
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('.StripeElementsForm').exists()).toEqual(true)
  expect(wrapper.find('#stripe-form').exists()).toEqual(true)
})

it('should disable submit button by default', () => {
  const wrapper = mount(<StripeElementsForm { ...props } />)
  expect(wrapper.find('#stripe-submit').prop('disabled')).toEqual(true)
})

// TODO: Update tests
// Testing @stripe/react-stripe-js is very tricky, in fact there are no official docs or examples
// Stripe team is working on that issue, see discussion in https://github.com/stripe/react-stripe-js/issues/59
// we should update this tests when the issue is updated
it.skip('should enable submit button when form is complete and valid', () => {
  const wrapper = mount(<StripeElementsForm { ...props } />)
  expect(wrapper.find('#stripe-submit').prop('disabled')).toEqual(true)
  wrapper.find('.CardNumberField-input-wrapper span input').simulate('change', { target: { value: '4111 1111 1111 1111' } })
  wrapper.find('.CardField-expiry span span .InputElement')
    .simulate('change', { target: { value: '11 / 22' } })
  wrapper.find('.CardField-cvc span span .InputElement')
    .simulate('change', { target: { value: '123' } })
  wrapper.find('.CardField-postalCode span span .InputElement')
    .simulate('change', { target: { value: '01234' } })
  expect(wrapper.find('#stripe-submit').prop('disabled')).toEqual(false)
})
