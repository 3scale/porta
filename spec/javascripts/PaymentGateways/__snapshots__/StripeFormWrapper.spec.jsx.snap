// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<StripeElementsForm
  billingAddressDetails={
    Object {
      "city": "South Park",
      "country": "US",
      "line1": "1002 Avenue de los Mexicanos",
      "line2": "",
      "postal_code": "80440",
      "state": "CO",
    }
  }
  isCreditCardStored={false}
  setupIntentSecret="efgh"
  stripePublishableKey="abcd"
  successUrl="/Broflovski/Residence"
>
  <Elements
    stripe={Promise {}}
  >
    <CardForm
      billingAddressDetails={
        Object {
          "city": "South Park",
          "country": "US",
          "line1": "1002 Avenue de los Mexicanos",
          "line2": "",
          "postal_code": "80440",
          "state": "CO",
        }
      }
      isCreditCardStored={false}
      setupIntentSecret="efgh"
      successUrl="/Broflovski/Residence"
    >
      <div
        className="well StripeElementsForm"
      >
        <EditCreditCardDetails
          isStripeFormVisible={true}
          onToogleVisibility={[Function]}
        >
          <a
            className="editCardButton"
            onClick={[Function]}
          >
            <i
              className="fa fa-chevron-left"
            />
            <span>
              cancel
            </span>
          </a>
        </EditCreditCardDetails>
        <form
          acceptCharset="UTF-8"
          action="/Broflovski/Residence"
          className=""
          id="stripe-form"
          method="post"
          onSubmit={[Function]}
        >
          <fieldset>
            <legend>
              Credit card details
            </legend>
            <CardElement
              className="col-md-12"
              onChange={[Function]}
              options={
                Object {
                  "classes": Object {
                    "empty": "is-empty",
                    "focus": "is-focused",
                  },
                  "iconStyle": "solid",
                  "style": Object {
                    "base": Object {
                      "::placeholder": Object {
                        "color": "#3f454c",
                      },
                      "color": "black",
                      "fontFamily": "\\"Helvetica Neue\\", Helvetica, sans-serif",
                      "fontSize": "19px",
                      "fontWeight": 300,
                      "iconColor": "#8898AA",
                    },
                    "invalid": Object {
                      "color": "#e85746",
                      "iconColor": "#e85746",
                    },
                  },
                }
              }
            >
              <div
                className="col-md-12"
              />
            </CardElement>
          </fieldset>
          <fieldset>
            <div
              className="form-group"
            >
              <div
                className="col-md-10 operations"
              >
                <button
                  className="btn btn-primary pull-right"
                  disabled={true}
                  id="stripe-submit"
                  type="submit"
                >
                  Save details
                </button>
              </div>
            </div>
          </fieldset>
          <input
            id="stripe_payment_method_id"
            name="stripe[payment_method_id]"
            type="hidden"
            value=""
          />
          <CSRFToken />
        </form>
      </div>
    </CardForm>
  </Elements>
</StripeElementsForm>
`;
